src/auth/strategies/magic-link.strategy.ts

```diff
@@ -24,44 +26,87 @@ export class MagicLinkStrategy extends PassportStrategy(
   }
 
   async authenticate(req: Request): Promise<void> {
-    const { authorization } = req.headers;
-    if (!authorization) {
-      return this.fail({
-        code: 'no_authorization',
-        message: 'Error: Request header is invalid.',
-      });
-    }
-
     if (this.config.get('DISABLE_LOGIN')) {
       return this.fail({
         code: 'disable_login',
         message: 'Error: Login is disabled.',
       });
     }
 
+    const { authorization } = req.headers;
+    const cookies = cookie.parse(req.headers.cookie || '');
+
     let email;
-    try {
-      email = await this.magicLinkService.getEmailFromHeader(authorization);
-    } catch {
-      return this.fail({
-        code: 'email_error',
-        message: 'Error: Failed to get the email. Please try again.',
-      });
+    if (authorization) {
+      try {
+        email = await this.magicLinkService.getEmailFromHeader(authorization);
+      } catch {
+        return this.fail({
+          code: 'email_error',
+          message: 'Error: Failed to get the email. Please try again.',
+        });
+      }
+    } else if (cookies) {
+      try {
+        const token: string = cookies.jwt;
+        const decoded = jwt.verify(token, this.config.get('JWT_TOKEN_SECRET'));
+
+        if (decoded.sub === undefined || decoded.sub.toString().length === 0) {
+          return this.fail({
+            code: 'invalid_token',
+            message: 'Error: Jwt token is missing email.',
+          });
+        }
+
+        email = decoded.sub.toString();
+      } catch (err) {
+        if (err instanceof JsonWebTokenError) {
```

Comment:
`JsonWebTokenError` has a `name` and `message`. Can we that message instead of wrapping it? I don't think we need to add anything more to that

Reply:
Updated, I did not use it at first because the name in `JsonWebTokenError` is camel case but our code format is snake case.